Sources/Core/Renderer/GUI/GUIElement/GUITextInput.swift

```diff
@@ -3,6 +3,7 @@ import Foundation
 struct GUITextInput: GUIElement {
   var content: String
   var width: Int
+  var cursorIndex: Int
```

Comment:
Once you address the suggestion at line 150 of `PlayerInputSystem.swift`, I reckon this index property should be assigned the value of `messageInputCursorIndex` (making it a `String.Index`) instead of `messageInputCursor`, since that will simplify the code a bit (this code doesn't benefit from the cursor being indexed backwards).